go/arrow/math/Makefile

```diff
@@ -37,27 +41,42 @@ INTEL_SOURCES := \
 	int64_avx2_amd64.s int64_sse4_amd64.s \
 	uint64_avx2_amd64.s uint64_sse4_amd64.s
 
+ARM_SOURCES := \
+	float64_neon_arm64.s \
+	int64_neon_arm64.s \
+	uint64_neon_arm64.s
+
 .PHONEY: assembly
 
-assembly: $(INTEL_SOURCES)
+#ifeq ($(GOARCH), arm64)
+assembly: $(ARM_SOURCES)
+#else
+#assembly: $(INTEL_SOURCES)
+#endif
 
```

Comment:
Does this only work if you're currently *on* arm64? I believe Go has arguments and env vars which allow for creating / building cross platform, is it possible to build the ARM sources for Arm64 neon without having to be on an arm64 neon machine? Either way, can we have comments or the readme updated with whatever instructions / requirements are needed to generate the arm64 sources?

Reply:
@zeroshade Thanks for your comments. :-)
Agreed on what you mentioned. 

But after further consideration, it's noted that `c2goasm`doesn't work correctly for Arm64. 
The Arm64 assembly were generated by `asm2plan9s`, and I manually formatted the related assembler codes.

IMO, it seems that we would not add the `assembly` target for Arm64 until `c2goasm` added the Arm64 support.  
The `ARM_SOURCES` should be removed.

Reply:
@guyuqi Sounds good to me, can we just document either in a comment or in the readme the procedure for re-generating the arm64 assembly including a comment about the fact that c2goasm doesn't support it? I'm happy once there's a comment or documentation in the readme just so we don't lose this info if we lose track of this PR.

Reply:
Agreed. I'll add a comment in assembly files to retain the related info to track this PR, thanks.

Reply:
Updated.